src/vaadin-grid.js

```diff
@@ -0,0 +1,930 @@
+ * `drag-disabled` | Set to a row that isn't available for dragging | row
+ * `drop-disabled` | Set to a row that can't be dropped on top of | row
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
+ *
+ * @extends ScrollerElement
+ * @mixes ElementMixin
+ * @mixes ThemableMixin
+ * @mixes A11yMixin
+ * @mixes ActiveItemMixin
+ * @mixes ArrayDataProviderMixin
+ * @mixes ColumnResizingMixin
+ * @mixes DataProviderMixin
+ * @mixes DynamicColumnsMixin
+ * @mixes FilterMixin
+ * @mixes RowDetailsMixin
+ * @mixes ScrollMixin
+ * @mixes SelectionMixin
+ * @mixes SortMixin
+ * @mixes KeyboardNavigationMixin
+ * @mixes ColumnReorderingMixin
+ * @mixes EventContextMixin
+ * @mixes StylingMixin
+ * @mixes DragAndDropMixin
+ */
+class GridElement extends ElementMixin(
+  ThemableMixin(
+    DataProviderMixin(
+      ArrayDataProviderMixin(
+        DynamicColumnsMixin(
+          ActiveItemMixin(
+            ScrollMixin(
+              SelectionMixin(
+                SortMixin(
+                  RowDetailsMixin(
+                    KeyboardNavigationMixin(
+                      A11yMixin(
+                        FilterMixin(
+                          ColumnReorderingMixin(
+                            ColumnResizingMixin(EventContextMixin(DragAndDropMixin(StylingMixin(ScrollerElement))))
+                          )
+                        )
+                      )
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+) {
+  static get template() {
+    return html`
+      <div
+        id="scroller"
+        safari$="[[_safari]]"
+        ios$="[[_ios]]"
+        loading$="[[loading]]"
+        column-reordering-allowed$="[[columnReorderingAllowed]]"
+      >
+        <table id="table" role="grid" aria-multiselectable="true" tabindex="0">
+          <caption id="sizer" part="row"></caption>
+          <thead id="header" role="rowgroup"></thead>
+          <tbody id="items" role="rowgroup"></tbody>
+          <tfoot id="footer" role="rowgroup"></tfoot>
+        </table>
+
+        <div part="reorder-ghost"></div>
+      </div>
+
+      <div id="focusexit" tabindex="0"></div>
+    `;
+  }
+
+  static get is() {
+    return 'vaadin-grid';
+  }
+
+  static get version() {
+    return '5.7.7';
+  }
+
+  static get observers() {
+    return ['_columnTreeChanged(_columnTree, _columnTree.*)'];
+  }
+
+  static get properties() {
+    return {
+      /** @private */
+      _safari: {
+        type: Boolean,
+        value: /^((?!chrome|android).)*safari/i.test(navigator.userAgent)
+      },
+
+      /** @private */
+      _ios: {
+        type: Boolean,
+        value:
+          (/iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream) ||
+          (navigator.platform === 'MacIntel' && navigator.maxTouchPoints > 1)
+      },
+
+      /** @private */
+      _edge: {
```

Comment:
This detects Edge legacy and can be removed.

Reply:
Done.